tasks/update.yml

```diff
@@ -0,0 +1,91 @@
+---
+# Creates a folder /opt/mattermost-updated/mattermost with the updated version
+
+- name: Remove a previously leaked mattermost-displaced folder
+  file:
+    path: /opt/mattermost-displaced
+    state: absent
+
+- name: Remove a previously leaked mattermost-updated folder
+  file:
+    path: /opt/mattermost-updated
+    state: absent
+
+- name: Create the displaced folder
+  file:
+    path: /opt/mattermost-displaced
+    state: directory
+
+- block:
+
+  - name: 'Download team version binary from https://releases.mattermost.com'
+    get_url:
+      url: 'https://releases.mattermost.com/{{ mattermost_version }}/mattermost-team-{{ mattermost_version }}-linux-amd64.tar.gz'
+      dest: "{{ global_cache_dir | mandatory }}/mattermost-team-{{ mattermost_version }}/"
+    become: false
+    delegate_to: localhost
+
+  - name: Unpack Mattermost team archive
+    unarchive:
+      src: "{{ global_cache_dir }}/mattermost-team-{{ mattermost_version }}-linux-amd64.tar.gz"
+      dest: /opt/mattermost-displaced/
+    args:
+
+  when: not mattermost_enterprise
+
+- block:
+
+  - name: 'Download enterprise version binary from https://releases.mattermost.com'
+    get_url:
+      url: 'https://releases.mattermost.com/{{ mattermost_version }}/mattermost-{{ mattermost_version }}-linux-amd64.tar.gz'
+      dest: "{{ global_cache_dir | mandatory }}/"
+    become: false
+    delegate_to: localhost
+
+  - name: Unpack Mattermost enterprise archive
+    unarchive:
+      src: "{{ global_cache_dir }}/mattermost-{{ mattermost_version }}-linux-amd64.tar.gz"
+      dest: /opt/mattermost-displaced/
+    args:
+
+  when: mattermost_enterprise
+
+- name: Make sure the mattermost folder exists
+  file:
+    path: /opt/mattermost
+    state: directory
+
+- name: Create preserved folders at the source
+  file:
+    path: /opt/mattermost/{{ item }}
+    state: directory
+    recurse: yes
+  with_items: '{{ mattermost_preserve_on_update }}'
+
+- name: Create preserved folders at the destination
+  file:
+    path: /opt/mattermost-displaced/mattermost/{{ item }}
```

Comment:
Is there a reason for the folder `mattermost`? Why not place the files directly in the displaced folder?

Reply:
The folder inside the downloaded archive is called `mattermost`